scalding-core/src/main/scala/com/twitter/scalding/Execution.scala

```diff
@@ -192,6 +193,8 @@ sealed trait Execution[+T] extends java.io.Serializable {
    */
   def zip[U](that: Execution[U]): Execution[(T, U)] =
     Zipped(this, that)
+
+  override val hashCode = super.hashCode
```

Comment:
Yeah, that's what I meant (not sure if "strict" means something other than eager/val).

And `ScalaRunTime._hashCode` seems to be exactly what we want here.  All of the subclasses are `Product` anyway; so maybe it could stay in the base trait too.